native/cocos/renderer/gfx-metal/MTLCommandBuffer.mm

```diff
@@ -319,15 +326,25 @@ of this software and associated engine source code (the "Software"), a limited,
 
     const DepthStencilAttachment& dsAttachment  = curRenderPass->getDepthStencilAttachment();
     const SubpassInfo&      subpass             = subpasses[index];
-    if (subpass.depthStencil != INVALID_BINDING) {
+    uint32_t ds = subpass.depthStencil;
+    auto iter = std::find_if(subpass.inputs.begin(), subpass.inputs.end(), [&subpass](uint32_t index){ return index >= subpass.inputs.size(); });
```

Comment:
Should it be `index >= subpass.colors.size();` ?

Reply:
subpass[0]:
-- input:    []
-- output:  [0, 1, 2]

subpass[1]:
-- input:    [0, 1, 2, 4]
-- output: [3]

to activate Metal Subpass, only 1 renderpassDescriptor is allowed, which means there will be 2 pass once any attachment in slot is changed(because this product another different renderpassDescriptor). So at the very beginning we have to set the slot in renderPassDescriptor: [0, 1, 2, 3, 4], that's why I look for the ds first in subpass.inputs and then in outputs.

Reply:
append: sth slightly different in Metal: no explicitly input and output attachment, there's just attachments lie in certain slot in renderpass descriptor. 
eg: renderPassDescriptor with slots: [0, 1, 2, 3, 4], in metal shader `[[colors[0]]]` get the attachment at 0, can be used as input OR output

Reply:
I see.